packages/netlify-cms-widget-markdown/src/MarkdownControl/plugins/CommandsAndQueries.js

```diff
@@ -71,6 +71,37 @@ function CommandsAndQueries({ defaultType }) {
       hasInline(editor, type) {
         return editor.value.inlines.some(node => node.type === type);
       },
+      hasQuote(editor, quoteLabel) {
+        const { value } = editor;
+        const { document, blocks } = value;
+        return blocks.some(node => {
+          const { key: descendantNodeKey } = node;
+          const parent = document.getClosest(
+            descendantNodeKey,
+            parentNode => parentNode.type === quoteLabel,
+          );
+          return !!parent;
+        });
+      },
+      hasListItems(editor, listType) {
+        const { value } = editor;
+        const { document, blocks } = value;
+        return blocks.some(node => {
+          const { key: lowestNodeKey, type: lowestNodeType } = node;
```

Comment:
Similar to the above comment this can mark a list item as both a bulleted list and a numbered list:
![image](https://user-images.githubusercontent.com/26760571/119352511-6fbc3680-bcaa-11eb-98a4-e78a2bed4526.png)

Would the following work:
```js
return blocks.some(node => {
  const parent = document.getParent(node.key);
  const grandParent = parent && document.getParent(parent.key);

  return (
    node.type === 'paragraph' &&
    parent?.type === 'list-item' &&
    grandParent?.type === listType
  );
});
```

Reply:
It works. Thanks for your suggestion, although I'm starting to feel like the lax schema in Netlify CMS' s will open up new unexpected UI problems in the future.

Reply:
How do you suggest we tackle the schema issue. We've been wanting to update slate to the latest version but it's quite a heavy lift. We could re-enable the tests here https://github.com/netlify/netlify-cms/blob/26cd552746d6294387b9bdd961a73d5a531bf665/packages/netlify-cms-widget-markdown/src/serializers/__tests__/commonmark.spec.js#L67
and see what's failing too

Reply:
I haven't had a concrete idea, but at the very least, `quote` and `list-item` blocks shouldn't have heading, code block, table etc as child nodes. The behavior of Backspace and Enter is also unpredictable. Is there a dedicated thread where we can discuss more about schema and document transformation in general?

Reply:
> Is there a dedicated thread where we can discuss more about schema and document transformation in general?

We have a few issues about our editor various quirks, but I don't think we have a central one. Would you like to open it and start a discussion?

Reply:
I've discovered another issue:

If I'm in a `list-item` block, then click any Heading button. Now, the List button will be dimmed and the Heading button highlighted.

![quote-block-inside-list](https://user-images.githubusercontent.com/28924121/119812879-d3e13380-bf12-11eb-95f9-b3601b070d9c.gif)

That's because one of the condition for active list is the lowest node has to be `paragraph` block. But now the lowest node has become a `heading` block, because the schema allows that.

Should I remove `node.type === 'paragraph'` condition? The downside is that both Heading and List buttons will be highlighted, which is potentially confusing to end users.

Reply:
> Should I remove `node.type === 'paragraph'` condition? The downside is that both Heading and List buttons will be highlighted, which is potentially confusing to end users.

I think removing `node.type === 'paragraph'` makes sense due to the current behavior.